lib/filter/filter-cmp.c

```diff
@@ -53,9 +53,16 @@ fop_compare_numeric(const GString *left, const GString *right)
 }
 
 static gint
-fop_compare_string(const GString *left, const GString *right)
+fop_compare_bytes(const GString *left, const GString *right)
 {
-  return strcmp(left->str, right->str);
+  gint cmp = memcmp(left->str, right->str, MIN(left->len, right->allocated_len));
```

Comment:
Why is this "allocated_len" on the right side? That's not obvious to me.

Reply:
I think this was caused of a wrong autocomplete that I missed. Nice catch! Fixed.